spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/config/Profiles.java

```diff
@@ -128,7 +130,7 @@ private boolean hasExplicit(Supplier<String[]> supplier, String propertyValue, S
 		}
 		List<String> reversed = new ArrayList<>(list);
 		Collections.reverse(reversed);
-		return Collections.unmodifiableList(reversed);
+		return reversed;
```

Comment:
Alternatively, we could do a stream/filter approach, but the unmodifiable list seemed unnecessary as the result is only used above and cannot be modified from the outside.